### ISSUE TITLE
xref versus namespace in multiple file

### ISSUE THREAD

**Issue by pailheres** [doxygen/doxygen#7403]
# Problem

- If a same namespace is defined in multiple C++ files
- If in one (or more) of those files, there is an xref in the comment (\todo, custom xref, etc)

Depending of the order in which the files get parsed, some xref list item can be missing.

xref are always present in the namespace docs, but not in the xref list

# Correction
Xref must be appended to the list instead of the list being overwritten when a namespace already existing is parsed. 

In the file [definition.cpp](https://github.com/doxygen/doxygen/blob/master/src/definition.cpp), if i change the following function, it correct the bug

```cpp
void DefinitionImpl::setRefItems(const std::vector<ListItemInfo> &sli)
{
  //an append is needed, not an overwite
  //m_impl->xrefListItems = sli;
  for (auto &blabla : sli)
  {
    m_impl->xrefListItems.push_back(blabla);
  }
}
```

**Comment by albert-github** [comment]
Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows shows us also the problem?  Please don't add external links as they might not be persistent.

**Comment by pailheres** [comment]
With this example code, the doc generated is ok with1.8.16, but not if i use latest doxygen from git

[example_bug_doxygen.zip](https://github.com/doxygen/doxygen/files/3852197/example_bug_doxygen.zip)

**Comment by albert-github** [comment]
Looks like a regression introduced in: Commit: 38f1441dc329e72ba4c782fdc721ce5199e01a67 [38f1441]
Commit Date: Tuesday, October 29, 2019 10:54:24 PM
Replaced QList<ListItemInfo> with std::vector<ListItemInfo>

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.17.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
